subprojects/testing-base/src/main/java/org/gradle/testing/base/internal/DefaultTestingExtension.java

```diff
@@ -16,26 +16,361 @@
 
 package org.gradle.testing.base.internal;
 
+import groovy.lang.Closure;
+import org.gradle.api.Action;
+import org.gradle.api.DomainObjectCollection;
 import org.gradle.api.ExtensiblePolymorphicDomainObjectContainer;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.NamedDomainObjectCollectionSchema;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.NamedDomainObjectFactory;
+import org.gradle.api.NamedDomainObjectProvider;
+import org.gradle.api.NamedDomainObjectSet;
+import org.gradle.api.Namer;
+import org.gradle.api.Rule;
+import org.gradle.api.UnknownDomainObjectException;
 import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.specs.Spec;
 import org.gradle.testing.base.TestSuite;
+import org.gradle.testing.base.TestSuiteContainerX;
 import org.gradle.testing.base.TestingExtension;
 
 import javax.inject.Inject;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
 
 public abstract class DefaultTestingExtension implements TestingExtension {
-    private final ExtensiblePolymorphicDomainObjectContainer<TestSuite> suites;
+    private final TestSuiteContainerX suites;
 
     @Inject
     public DefaultTestingExtension() {
-        this.suites = getObjectFactory().polymorphicDomainObjectContainer(TestSuite.class);
+        this.suites = new DelegatingTestSuiteContainerX(getObjectFactory().polymorphicDomainObjectContainer(TestSuite.class));
     }
 
     @Inject
     public abstract ObjectFactory getObjectFactory();
 
     @Override
-    public ExtensiblePolymorphicDomainObjectContainer<TestSuite> getSuites() {
+    public TestSuiteContainerX getSuites() {
         return suites;
     }
+
+    private static class DelegatingTestSuiteContainerX implements TestSuiteContainerX {
```

Comment:
Other containers extend the concrete container type. I think we should do it the same way here.
This would change how the container is instantiated.